lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1704,8 +1763,10 @@ std::vector<SPIRVWord> getBankBitsFromString(StringRef S) {
 void addIntelFPGADecorations(
     SPIRVEntry *E,
     std::vector<std::pair<Decoration, std::string>> &Decorations) {
-  if (!E->getModule()->isAllowedToUseExtension(
-          ExtensionID::SPV_INTEL_fpga_memory_attributes))
+  SPIRVModule *M = E->getModule();
+  if (!M->isAllowedToUseExtension(
```

Comment:
Note: I guess that in our stack both extensions most likely will be enabled/disabled together, but strictly speaking, this `if` is not fully correct: it will allow to use functionality from `SPV_INTEL_fpga_memory_attributes` even if it is disabled, but `SPV_INTEL_fpga_memory_accesses` enabled

Reply:
I see. I don't know if there's a quick way to separate these conditions, but at the very least, I should add a `TODO` for addressing this.

Reply:
I suppose that solution is to use || instead of &&.

Reply:
Oh, no, this check is for all decorations. Forget it.

Reply:
I've entirely removed the check from the function, as the same check is performed at caller site. The corresponding `TODO` was added to reflect the issue.